### ISSUE TITLE
wxIsBusy’ was not declared in this scope

### ISSUE THREAD

**Issue by simsong** [scottdraves/electricsheep#50]
I'm trying to compile electricsheep on Raspberry PI. 
```
PRETTY_NAME="Raspbian GNU/Linux 8 (jessie)"
NAME="Raspbian GNU/Linux"
VERSION_ID="8"
VERSION="8 (jessie)"
ID=raspbian
ID_LIKE=debian
```

I installed all of the packages and am getting this error:

```
In file included from /usr/include/wx-3.0/wx/window.h:22:0,
                 from /usr/include/wx-3.0/wx/sizer.h:16,
                 from config.h:21,
                 from config.cpp:8:
/usr/include/wx-3.0/wx/cursor.h: In constructor ‘wxBusyCursorSuspender::wxBusyCursorSuspender()’:
/usr/include/wx-3.0/wx/cursor.h:93:22: error: ‘wxIsBusy’ was not declared in this scope
         if( wxIsBusy() )
                      ^
/usr/include/wx-3.0/wx/cursor.h:95:26: error: ‘wxBusyCursor’ has not been declared
             wxSetCursor( wxBusyCursor::GetStoredCursor() );
                          ^
/usr/include/wx-3.0/wx/cursor.h: In destructor ‘wxBusyCursorSuspender::~wxBusyCursorSuspender()’:
/usr/include/wx-3.0/wx/cursor.h:100:22: error: ‘wxIsBusy’ was not declared in this scope
         if( wxIsBusy() )
                      ^
/usr/include/wx-3.0/wx/cursor.h:102:26: error: ‘wxBusyCursor’ has not been declared
             wxSetCursor( wxBusyCursor::GetBusyCursor() );
                          ^
In file included from /usr/include/wx-3.0/wx/window.h:25:0,
                 from /usr/include/wx-3.0/wx/sizer.h:16,
                 from config.h:21,
                 from config.cpp:8:
/usr/include/wx-3.0/wx/region.h: In member function ‘bool wxRegionBase::Intersect(const wxRect&)’:
/usr/include/wx-3.0/wx/region.h:242:37: error: invalid use of incomplete type ‘class wxRegion’
     return DoIntersect(wxRegion(rect));
                                     ^
In file included from /usr/include/wx-3.0/wx/bitmap.h:19:0,
                 from config.h:11,
                 from config.cpp:8:
/usr/include/wx-3.0/wx/gdicmn.h:37:28: error: forward declaration of ‘class wxRegion’
 class WXDLLIMPEXP_FWD_CORE wxRegion;
                            ^
In file included from /usr/include/wx-3.0/wx/window.h:25:0,
                 from /usr/include/wx-3.0/wx/sizer.h:16,
                 from config.h:21,
                 from config.cpp:8:
/usr/include/wx-3.0/wx/region.h: In member function ‘bool wxRegionBase::Subtract(const wxRect&)’:
/usr/include/wx-3.0/wx/region.h:247:36: error: invalid use of incomplete type ‘class wxRegion’
     return DoSubtract(wxRegion(rect));
--M
```
Well, it goes on for a few hundred pages.

**Comment by scattass** [comment]
I also got this same error trying to compile on debian 8. It seems like it may be wxwidget changed a lot since 2.9 which the configure file calls for. I'm not very adept at this stuff though so I could be wrong.

**Comment by scattass** [comment]
I was having this same issue and found this to work on debian 8. 

sudo apt-get install subversion autoconf libtool libgtk2.0-dev libgl1-mesa-dev libavcodec-dev libavformat-dev libswscale-dev liblua5.1-0-dev libcurl4-openssl-dev libxml2-dev libjpeg62-turbo-dev libboost-dev libboost-filesystem-dev libboost-thread-dev libtinyxml-dev freeglut3-dev glee-dev

sudo apt-get install libwxgtk3.0-dev

sudo apt-get install flam3


the only difference from the wiki compile guide being using the wxwidget development library from the debian repo (maybe it's patched?) and libjpeg62-turbo-dev instead of the previous one 8 which I assume is deprecated package.

**Comment by simsong** [comment]
I got it to compile using wxwidgets downloaded and compiled from source, rather than the
package manager.


----
Sent from my phone.

On May 19, 2017, at 3:11 AM, scattass <notifications@github.com> wrote:

I also got this same error trying to compile on debian 8. It seems like it may be wxwidget changed a lot since 2.9 which the configure file calls for. I'm not very adept at this stuff though so I could be wrong.
